nautobot/extras/filters/mixins.py

```diff
@@ -177,49 +177,32 @@ def __init__(self, side, relationship=None, queryset=None, qs=None, *args, **kwa
         self.side = side
         super().__init__(queryset=queryset, *args, **kwargs)
 
+    def generate_query(self, value, **kwargs):
```

Comment:
This change was needed to fix an issue when defining DynamicGroups with relationship filters. In the "old" code, such filters would work fine for filter-based groups, but would (previously unknown/unreported) cause errors if such a group was included in a set-based parent group. With the different code paths introduced by this refactor, even filter-based groups would demonstrate this issue, which was caught by one of our unit tests.